engineyard/ey-cookbooks-stable-v5
# [CC-1226] Ruby version updates

## Description of your patch

Ruby security updates to versions:
  * Ruby 2.2.9
  * Ruby 2.3.6
  * Ruby 2.4.3

## Recommended Release Notes

None.

## Estimated risk

Medium. 

## Components involved

Ruby 2.2, 2.3 and 2.4 packages

## Description of testing done

See QA instructions

## QA Instructions

1. Create a simple Ruby application. Tweaked version of TO-DO app placed (git@github.com:jegasega/todo.git). Displaying actual Ruby and Gems versions was added to the Environment info.
2. Boot the environment with the latest V5 stack version. Environment should be run with following parameters:
  * Application server - Passenger5
  * Runtime - Ruby 2.2
  * Environment type - Single Instance.
3. Visit your application to ensure it works correctly.
4. Edit your environment enable Development stacks and change stack version to the js_cc1226_ruby-v5 0.1.
5. Apply changes to your environment.
6. Redeploy application.
7. Visit your application to ensure it works correctly and Ruby version was updated to the 2.2.9
8. Change environment Ruby version to 2.3.
9. Apply changes to the environment.
10. Redeploy application.
11. Visit your application to ensure it works correctly and Ruby version was updated to the 2.3.6
12. Ruby version 2.4 still not available from the dashboard. For doing that we need manually update `/etc/chef/dna.json` with the information provided below

`"dna": {
...

        "ruby_version": "Ruby 2.4"
...`
and

`...
"components": [

          {
            "key": "ruby_240"
          },
          {
            "key": "rubygems",
            "version": "2.6.14"
          },

...`
11. Run Chef `PATH=/usr/local/ey_resin/bin:$PATH /home/ey/bin/chef-solo -j /etc/chef/dna.json -c /etc/chef/solo.rb`
12. Redeploy application
13. Visit your application to ensure it works correctly, Ruby version was changed to 2.4.3 and Ruby Gems version are 2.6.14.
